File: test_cafekiosk/src/main/java/com/testcafekiosk/spring/api/controller/order/OrderController.java
//package com.testcafekiosk.spring.api.controller.order;
//
//import com.testcafekiosk.spring.api.controller.order.request.OrderCreateRequest;
//import com.testcafekiosk.spring.api.service.order.OrderService;
//import com.testcafekiosk.spring.api.service.order.response.OrderResponse;
//import lombok.RequiredArgsConstructor;
//import org.springframework.web.bind.annotation.PostMapping;
//import org.springframework.web.bind.annotation.RequestBody;
//import org.springframework.web.bind.annotation.RestController;
//
//import java.time.LocalDateTime;
//
//@RestController
//@RequiredArgsConstructor
//public class OrderController {
//
//    private final OrderService orderService;
//
//    @PostMapping("/api/v1/orders/new")
//    public OrderResponse createOrder(@RequestBody OrderCreateRequest request) {
//        LocalDateTime registeredDateTime = LocalDateTime.now();
//        return orderService.createOrder(request, registeredDateTime);
//    }
//}

File: test_cafekiosk/src/main/kotlin/com/testcafekiosk/spring/api/controller/order/OrderController.kt
package com.testcafekiosk.spring.api.controller.order

import com.testcafekiosk.spring.api.ApiResponse
import com.testcafekiosk.spring.api.controller.order.request.OrderCreateRequest
import com.testcafekiosk.spring.api.service.order.OrderService
import com.testcafekiosk.spring.api.service.order.response.OrderResponse
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import java.time.LocalDateTime

@RestController
class OrderController(
    private val orderService: OrderService,
) {

    @PostMapping("/api/v1/orders/new")
    fun createOrder(@Valid @RequestBody request: OrderCreateRequest): ApiResponse<OrderResponse> {
        val registeredDateTime = LocalDateTime.now()
        return ApiResponse.ok(orderService.createOrder(request.toServiceRequest(), registeredDateTime))
    }
}